test/integration/test-amp-inabox.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {RequestBank} from '../../testing/test-helper';
+import {parseQueryString} from '../../src/url';
+
+describe.configure().skipIfPropertiesObfuscated().run('inabox', function() {
+  this.timeout(15000);
+
+  describes.integration('AMPHTML ads rendered on non-AMP page ATF', {
+    amp: false,
+    body: `
+      <iframe
+      src="//ads.localhost:9876/amp4test/inabox/${RequestBank.getBrowserId()}"
+          scrolling="no"
+          width="300" height="250">
+      </iframe>
+      <script src="/examples/inabox-tag-integration.js"></script>
```

Comment:
(low-key watching this PR because I'm looking to take advantage of it)

You may already be aware of this issue (if so sorry!), but due to #9120 the iframe will attempt to get visibility information from the top level window, but since in Karma the top-level window is not the test fixture with the host script loaded, the message will be discarded without results. The tests would still pass on browsers that didn't need the host script in the first place, but it fails on Safari due to lacking IntersectionObserver.

Reply:
@zombifier that's a good observation, that might be the cause of the Safari failing. I haven't looked into it yet (was OOO for a couple of days). Will update this thread once I figured out.